src/Resources/src/stores/dishesStore.ts

```diff
@@ -0,0 +1,236 @@
+     */
+    function setFilter(filterStr: string) {
+        DishesState.filter = filterStr;
+    }
+
+    /**
+     * Returns a list of dishes that contain the filter string in their title, in the title of their variations or in the title of their category
+     */
+    const filteredDishes = computed(() => {
+        const { getCategoryIdsByTitle } = useCategories();
+        return DishesState.dishes.filter(dish => dishContainsString(dish, DishesState.filter) || getCategoryIdsByTitle(DishesState.filter).includes(dish.categoryId));
+    });
+
+    /**
+     * Determines wether a dish contains the search string in its title or in the title of one of its variations
+     */
+    function dishContainsString(dish: Dish ,searchStr: string) {
+        return (
+            dish.titleDe.toLowerCase().includes(searchStr.toLowerCase())
+            || dish.titleEn.toLowerCase().includes(searchStr.toLowerCase())
+            || dish.variations.map(variation => dishContainsString(variation, searchStr)).includes(true)
+        );
+    }
+
+    /**
+     * Fetches a list of dishes from the API and updates the DishesState
+     */
+    async function fetchDishes() {
+        DishesState.isLoading = true;
+
+        const { dishes, error } = await getDishes();
+        if (!error.value && dishes.value) {
+            DishesState.dishes = dishes.value;
+            DishesState.error = '';
+        } else {
+            DishesState.error = 'Error on fetching dishes';
+            setTimeout(fetchDishes, TIMEOUT_PERIOD);
+        }
+
+        DishesState.isLoading = false;
+    }
+
+    /**
+     * Calls postCreateDish to create a dish and fetches the dishes again
+     * @param dish The dish to create
+     */
+    async function createDish(dish: CreateDishDTO) {
+        const { error, response } = await postCreateDish(dish);
+
+        if (error.value || response.value?.status !== 'success') {
+            DishesState.error = 'Error on creating dish';
+            return;
+        }
+
+        await fetchDishes();
+    }
+
+    /**
+     * Calls deleteDish to delete a dish and fetches the dishes again
+     * @param slug The slug of the dish to delete
+     */
+    async function deleteDishWithSlug(slug: string) {
+        const { error, response } = await deleteDish(slug);
+
+        if (error.value || response.value?.status !== 'success') {
+            DishesState.error = 'Error on creating dish';
+            return;
+        }
+
+        await fetchDishes();
+    }
+
+    /**
+     * Creates a dishVariation and fetches the dishes again
+     * @param dishVariation The dishVariation to create
+     * @param parentSlug The identifier of the parent dish
+     */
+    async function createDishVariation(dishVariation: CreateDishVariationDTO, parentSlug: string) {
+        const { error, response } = await postCreateDishVariation(dishVariation, parentSlug);
+
+        if (error.value || response.value?.status !== 'success') {
```

Comment:
diese drei zeilen gibt es in ähnlicher form mehrfach in dieser datei, auslagern in eine funktion?